src/db/oauth_db.rs

```diff
@@ -0,0 +1,132 @@
+use std::borrow::Cow;
+
+use authly_common::id::DirectoryId;
+use authly_db::{param::AsParam, Db, DbResult, FromRow};
+use hiqlite::{params, Param, Params};
+use indoc::indoc;
+
+use crate::{directory::OAuthDirectory, encryption::DecryptedDeks, id::BuiltinProp};
+
+use super::cryptography_db::{CrDbError, EncryptedObjIdent};
+
+pub fn upsert_oauth_directory_stmt(
```

Comment:
Maybe the _directory type_ is called "OpenID Connect" - it provides identities, but Authly uses an OAuth API to interface with the provider. That way the OIC + OAuth duality will be nicely represented..